app/src/main/java/org/thoughtcrime/securesms/service/ExpiringMessageManager.java

```diff
@@ -54,10 +55,10 @@ public void checkSchedule() {
 
   private class LoadTask implements Runnable {
     public void run() {
-      SmsDatabase.Reader smsReader = smsDatabase.readerFor(smsDatabase.getExpirationStartedMessages());
-      MmsDatabase.Reader mmsReader = mmsDatabase.getExpireStartedMessages(null);
+      SmsDatabase.Reader smsReader = SmsDatabase.readerFor(smsDatabase.getExpirationStartedMessages());
+      MmsDatabase.Reader mmsReader = MmsDatabase.readerFor(mmsDatabase.getExpirationStartedMessages());
 
-      MessageRecord messageRecord = null;
+      MessageRecord messageRecord;
 
       while ((messageRecord = smsReader.getNext()) != null) {
         expiringMessageReferences.add(new ExpiringMessageReference(messageRecord.getId(),
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.



**Problem**  
 This line may not be thread-safe: **64**. At this line the following field is not protected by synchronization: `expiringMessageReferences`. Your code typically protects this field with synchronization on object: `expiringMessageReferences`.

**Fix**  
 Consider protecting the above field with synchronization on object: `expiringMessageReferences`. For example, like at lines: **52**, **45**, and **46**.

**More info**  
 The following lines are where the above field is already protected by synchronization on the above object: **45**, **46**, **52**, **88**, **88**, **90**, **95**, and **98**.